test_tf2/test/test_message_filter.cpp

```diff
@@ -68,37 +64,54 @@ class Notification
 
 TEST(MessageFilter, noTransforms)
 {
-  BufferCore bc;
+  auto node = rclcpp::Node::make_shared("tf2_ros_message_filter");
+  auto create_timer_interface = std::make_shared<tf2_ros::CreateTimerROS>(
+    node->get_node_base_interface(),
+    node->get_node_timers_interface());
+
+  rclcpp::Clock::SharedPtr clock = std::make_shared<rclcpp::Clock>(RCL_SYSTEM_TIME);
+  tf2_ros::Buffer buffer(clock);
+  buffer.setCreateTimerInterface(create_timer_interface);
+  tf2_ros::MessageFilter<geometry_msgs::msg::PointStamped> filter(buffer, "frame1", 10, node);
   Notification n(1);
-  MessageFilter<geometry_msgs::PointStamped> filter(bc, "frame1", 1, 0);
-  filter.registerCallback(boost::bind(&Notification::notify, &n, _1));
+  filter.registerCallback(std::bind(&Notification::notify, &n, std::placeholders::_1));
 
-  geometry_msgs::PointStampedPtr msg(new geometry_msgs::PointStamped);
-  msg->header.stamp = builtin_interfaces::msg::Time(1);
+  std::shared_ptr<geometry_msgs::msg::PointStamped> msg = std::make_shared<geometry_msgs::msg::PointStamped>();
+  msg->header.stamp = tf2_ros::toMsg(tf2::timeFromSec(1));
   msg->header.frame_id = "frame2";
   filter.add(msg);
 
   EXPECT_EQ(0, n.count_);
+
+  node.reset();
```

Comment:
Are these calls to `reset` (here and elsewhere) necessary?  `node` goes out-of-scope at the end of this function, so I would think that it would automatically get destroyed at the end here.